Calibration/HcalCalibAlgos/plugins/HcalHBHEMuonAnalyzer.cc

```diff
@@ -281,265 +295,337 @@ void HcalHBHEMuonAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSet
-	Tight_LongPara.push_back(fabs(RecMuon->muonBestTrack()->dz((*firstGoodVertex).position())));
+	globalTrack_.push_back(true);
+	chiGlobal_.push_back(RecMuon->globalTrack()->normalizedChi2());
+	globalMuonHits_.push_back(RecMuon->globalTrack()->hitPattern().numberOfValidMuonHits());
+	matchedStat_.push_back(RecMuon->numberOfMatchedStations());  
+	globalTrckPt_.push_back(RecMuon->globalTrack()->pt());
+	globalTrckEta_.push_back(RecMuon->globalTrack()->eta());
+	globalTrckPhi_.push_back(RecMuon->globalTrack()->phi()); 
+	tight_TransImpara_.push_back(fabs(RecMuon->muonBestTrack()->dxy((*firstGoodVertex).position())));
+	tight_LongPara_.push_back(fabs(RecMuon->muonBestTrack()->dz((*firstGoodVertex).position())));
       } else {
-	GlobalTrack.push_back(false);
-	chiGlobal.push_back(0);
-	GlobalMuonHits.push_back(0);
-	MatchedStat.push_back(0);
-	GlobalTrckPt.push_back(0);
-	GlobalTrckEta.push_back(0);
-	GlobalTrckPhi.push_back(0);
-	Tight_TransImpara.push_back(0);
-	Tight_LongPara.push_back(0);
+	globalTrack_.push_back(false);
+	chiGlobal_.push_back(0);
+	globalMuonHits_.push_back(0);
+	matchedStat_.push_back(0);
+	globalTrckPt_.push_back(0);
+	globalTrckEta_.push_back(0);
+	globalTrckPhi_.push_back(0);
+	tight_TransImpara_.push_back(0);
+	tight_LongPara_.push_back(0);
       }
       
-      IsolationR04.push_back(((RecMuon->pfIsolationR04().sumChargedHadronPt + std::max(0.,RecMuon->pfIsolationR04().sumNeutralHadronEt + RecMuon->pfIsolationR04().sumPhotonEt - (0.5 *RecMuon->pfIsolationR04().sumPUPt))) / RecMuon->pt()) );
+      isolationR04_.push_back(((RecMuon->pfIsolationR04().sumChargedHadronPt + std::max(0.,RecMuon->pfIsolationR04().sumNeutralHadronEt + RecMuon->pfIsolationR04().sumPhotonEt - (0.5 *RecMuon->pfIsolationR04().sumPUPt))) / RecMuon->pt()) );
       
-      IsolationR03.push_back(((RecMuon->pfIsolationR03().sumChargedHadronPt + std::max(0.,RecMuon->pfIsolationR03().sumNeutralHadronEt + RecMuon->pfIsolationR03().sumPhotonEt - (0.5 * RecMuon->pfIsolationR03().sumPUPt))) / RecMuon->pt()));
+      isolationR03_.push_back(((RecMuon->pfIsolationR03().sumChargedHadronPt + std::max(0.,RecMuon->pfIsolationR03().sumNeutralHadronEt + RecMuon->pfIsolationR03().sumPhotonEt - (0.5 * RecMuon->pfIsolationR03().sumPUPt))) / RecMuon->pt()));
 
-      MuonEcalEnergy.push_back(RecMuon->calEnergy().emS9);		 
-      MuonHcalEnergy.push_back(RecMuon->calEnergy().hadS9);
-      MuonHOEnergy.push_back(RecMuon->calEnergy().hoS9);
+      ecalEnergy_.push_back(RecMuon->calEnergy().emS9);		 
+      hcalEnergy_.push_back(RecMuon->calEnergy().hadS9);
+      hoEnergy_.push_back(RecMuon->calEnergy().hoS9);
       
-      double eEcal(0),eHcal(0),activeL(0),eHcalDepth[7],eHcalDepthHot[7];
-      bool tmpmatch = false;
-      unsigned int isHot = 0;
-      for (int i=0; i<7; ++i) eHcalDepth[i]=eHcalDepthHot[i]=-10000;
+      double eEcal(0), eHcal(0), activeLengthTot(0), activeLengthHotTot(0);
+      double eHcalDepth[7], eHcalDepthHot[7], activeL[7], activeHotL[7];
```

Comment:
@bsunanda please use a named constant to clarify where the magic number 7 comes from or just use the size of the `std::array` proposed above